firmware/arduino/chessboard/circleFunction.ino

```diff
@@ -0,0 +1,250 @@
+int pulsesPerSlope[NUM_CIRCLES][NUM_SLOPES_PER_QUARTER_CIRCLE * 2 - 2];
+
+enum SlopeToIndex
+{
+    SLOPE_HORIZONTAL_MAX_SIZE = 0,
+    SLOPE_HORIZONTAL = 1,
+    SLOPE_EIGHTH = 2,
+    SLOPE_QUARTER_MAX_SIZE = 3,
+    SLOPE_QUARTER = 4,
+    SLOPE_HALF_MAX_SIZE = 5,
+    SLOPE_HALF = 6,
+    SLOPE_ONE_MAX_SIZE = 7,
+    SLOPE_ONE = 8,
+    SLOPE_TWO_MAX_SIZE = 9,
+    SLOPE_TWO = 10,
+    SLOPE_FOUR_MAX_SIZE = 11,
+    SLOPE_FOUR = 12,
+    SLOPE_EIGHT = 13,
+    SLOPE_VERTICAL_MAX_SIZE = 14,
+    SLOPE_VERTICAL = 15
+};
+
+
+// Makes a full circle of the given size (0=largest, NUM_CIRCLES-1=smallest) 
+// starting from the given quadrant (0=top, 1=left, 2=bottom, 3=right).
+// calculatePulsesPerSlope() must be called before makeCircle()
+void makeCircle(int circle, int firstQuarter)
+{
+    bool firstPass = true;
+    int slopeIndex;
+
+    setScale(xMotor, EIGHTH_STEPS);
+    digitalWrite(xMotor[STEP_PIN], LOW);
+
+    for (int quarter = firstQuarter; quarter != firstQuarter || firstPass; quarter = (quarter + 1) % 4)
+    {
+        firstPass = false;
+
+        // Set Y-direction
+        if (quarter == 0 || quarter == 1)
```

Comment:
Can you pls make preprocessors for these `quarter == blah` to make it more readable? e.g.
```
#define TOP = 0
```
etc.

Reply:
I added these as an enum to keep it in the same file, and with a QUARTER_ prefix to avoid conflict with the existing LEFT and RIGHT macros.